subprojects/packagefiles/pcre2/meson.build

```diff
@@ -160,13 +160,25 @@ if get_option('test') and not meson.is_cross_build() # wine wrappers are not bat
     c_args: config_h_defs + ['-DHAVE_CONFIG_H'],
     link_args: link_args)
 
-  if host_machine.system() == 'windows'
+  if c_compiler.get_argument_syntax() == 'msvc'
     runtest = find_program('RunTest.bat')
   else
     runtest = find_program('RunTest')
   endif
 
   test('RunTest', runtest,
+    should_fail: host_machine.system() == 'windows' and c_compiler.get_argument_syntax() != 'msvc',
+    env : ['srcdir=@0@'.format(meson.current_source_dir())],
+    workdir: meson.current_build_dir())
+
+  if c_compiler.get_argument_syntax() == 'msvc'
+    rungreptest = find_program('RunGrepTest.bat')
+  else
+    rungreptest = find_program('RunGrepTest')
+  endif
```

Comment:
TBH I'd put that in the same if/else that finds `runtest`.

Reply:
you mean

```
if get_option('test') and not meson.is_cross_build()
```

?

Reply:
```meson
  if c_compiler.get_argument_syntax() == 'msvc'
    runtest = find_program('RunTest.bat')
  else
    runtest = find_program('RunTest')
  endif


  if c_compiler.get_argument_syntax() == 'msvc'
    rungreptest = find_program('RunGrepTest.bat')
  else
    rungreptest = find_program('RunGrepTest')
  endif
```

Why not just
```meson
  if c_compiler.get_argument_syntax() == 'msvc'
    runtest = find_program('RunTest.bat')
    rungreptest = find_program('RunGrepTest.bat')
  else
    runtest = find_program('RunTest')
    rungreptest = find_program('RunGrepTest')
  endif
```